common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -42,28 +42,44 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
-  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String)
-  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube")
-  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo")
+
+  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String, customAttributes: String => Map[String, String]){
+    val commonVideoAttributes = Map(("width", "5"), ("height", "3"), ("layout", "responsive"))
+    def getAttributes = customAttributes(videoId) ++ commonVideoAttributes
+  }
+
+  def standardAttributes(videoId: String): Map[String,String] = { Map(("data-videoid", videoId)) }
+
+  def facebookAttributes(videoId: String): Map[String, String] = {
+    Map(("data-href", s"https://www.facebook.com/theguardian/videos/$videoId"), ("data-embed-as", "video"))
+  }
+
+
+
+  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube", customAttributes = id => standardAttributes(id))
```

Comment:
`case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube", standardAttributes(videoId))`

and

`sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String, customAttributes: Map[String, String])`

...would be cleaner.  It was probably my original suggestion that led you to this, my bad!